### PR TITLE
[windows] Distinguish between host/native download/tools for windows build

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Currently the precompiled tools for windows builds are x86/win32. For now stick with this assumption and instead of putting these tools into the target arch folder, put them into the win32 folder to keep arch libs/exes in correct arch folders.

## Motivation and context
The ability to build a build tool (Texturepacker) for windows. To build a target using win32 libs, we need to be able to add its dependencies to the 0_package.native-win32.list and some of the Texturepacker dependencies overlap with the core Kodi build dependencies. We cant distinguish between arch libs, so we have to keep them in separate locations and use the appropriate lib for the appropriate targets.

Currently we only use win32 build tools, but someday there may be a desire to use x86_64 (or heaven forbid arm/aarch64) as true build tools. I dont want to overcomplicate things for what ifs at this stage, so this is the simplest, but just enforces some assumptions in both the batch scripts and the cmake project

## How has this been tested?
build windows

## What is the effect on users?
N/A
## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by CrystalP** [comment]
I tried many times building win32, but ffmpeg kept failing. Actually seems broken in master as well, so unrelated to this PR, but a problem for good testing. Is it cached on the build-bots maybe?
image

x64 built like a charm the first time. Haven't tried the uwp variants, should I?

Are you sure about removing the option of x64 native tools at this point? Though I don't know your plans for the next steps.

**Comment by thexai** [comment]
Tested build win32, x64 and UWP-64 and all seems fine.

@CrystalP : make sure execute `download-dependencies.bat ` **before** build FFmpeg
(in this case from `...kodi\tools\buildsteps\windows\win32`)

`ERROR: libdav1d not found` if because is not download yet

I'm not sure though if correct place for these "common tools" is 

`...\kodi\project\BuildDependencies\win32\tools`

or already existent

`...\kodi\project\BuildDependencies\tools`

**Comment by CrystalP** [comment]
That's weird, because I used a fresh git worktree and "developer command prompt", and followed the normal steps.
Tried again, and of course it worked fine this time.

Maybe a mirror problem? the output of download-dependencies flashes by quickly, I didn't notice anything but wasn't paying much attention.

**Review by CrystalP — APPROVED** [review]
it builds.

File: tools/buildsteps/windows/vswhere.bat [hunk 1]
```diff
@@ -27,20 +27,14 @@ SET vcvars=no
 SET sdkver=
 
 SET vsver=
-SET toolsdir=%arch%
+REM Current tools are only using x86/win32
+REM ToDo: allow to set NATIVEPLATFORM to allow native tools based on actual native arch (eg x86/x86_64/arm/arm64)
+SET toolsdir=win32
```
Comment (CrystalP) [inline tools/buildsteps/windows/vswhere.bat:32]: Want some code to find host arch? NATIVEPLAFORM will be similar.

The cmake side would have to reflect the same logic though, don't know what to do for NATIVEPREFIX

Comment (fuzzard) [reply tools/buildsteps/windows/vswhere.bat:32]: Currently, no. Long term sure. I'd rather get something working with the assumptions already in place in the codebase than extend it out to something that really doesn't matter currently

Comment (CrystalP) [reply tools/buildsteps/windows/vswhere.bat:32]: OK. Picking one is fine but win32 is making things more complicated though, with 2 different cross-builds. The typical x64 host would target win32 when building the tools and the target arch for Kodi. Are there many win32 build hosts left.

Comment (fuzzard) [reply tools/buildsteps/windows/vswhere.bat:32]: https://github.com/xbmc/xbmc/blob/master/project/BuildDependencies/scripts/0_package.native-win32.list

This is the current assumption in our code base.  Whilst these are x86, there is no reason to go out of our way to have multiple different arch types for build tools.

**Comment by fuzzard** [comment]
> I'm not sure though if correct place for these "common tools" is
> 
> `...\kodi\project\BuildDependencies\win32\tools`
> 
> or already existent
> 
> `...\kodi\project\BuildDependencies\tools`
> 
> Maybe better move all to same place.
> 
> **EDIT:** Seems `...\kodi\project\BuildDependencies\tools` is old folder I have locally but not used in current build scripts so ignore latest comment.

This is still one thing im not entirely sure on the best path. One thing i know from the apple world, keeping things clear and distinct for arch type and purpose (ie native/target) makes things easier. This is why i went with dumping them into the win32 folder by default here. If we ever use x86_64 bit tools (or heaven forbid arm/aarch64), having them in their own arch folder makes it a clear distinction, and a way to easily know exactly what this tool is built for.

Now the reason why im contemplating doing a tools folder, our current download scripts will be split into two actions (native/target). What if native == target (ie win32 host, win32 target). Currently with our scripts, we fail horribly if the same dependency is in both lists (ie https://jenkins.kodi.tv/job/WIN-32/29757/console).
So im not entirely sure where i'll go with it right now, but splitting native dependencies from target dependencies makes it easier with our existing download scripts, but im thinking im still better to fix the download scripts to handle download/extract of the same dep twice. Otherwise we end up with potentially folders of win 32 tools, win 32 target, even though they are the exact same dependencies

For reference, this is a build that builds TexturePacker for all platforms when appropriate (ie the goal of all this) https://jenkins.kodi.tv/view/All/job/BuildMulti-Choice/1104/

File: tools/buildsteps/windows/vswhere.bat (hunk 1 above)
Comment (fuzzard) [reply tools/buildsteps/windows/vswhere.bat:32]: I reread earlier in this thread, and i think you have an incorrect idea of the existing system

> Are you sure about removing the option of x64 native tools at this point? Though I don't know your plans for the next steps.

There are no x64 native tools. The current system downloads and extracts the x86 tools into whatever target arch folder (x86/x86_64/uwp_arm/uwp_x64 etc). The tools arent any of those arch types, they are only x86.

**Comment by CrystalP** [comment]
folder per arch makes sense. problem is solvable.

It's getting stuck because it's not expecting the extracted folder to already exist in the scripts/tmp folder.
Output is redirected to null in script, but you get this when running manually:
image

The folder looks like this at that stage:
image


A few ways to improve.
1. Alter 7za command to %ZIP% x -y %1 to answer yes to all questions
2. clean everything in the tmp folder after each extraction, not only the archive file (Post-Cleaning step of the script)
3. test the presence of the already extracted folder and skip extraction

Last option requires knowledge of the folder name inside the .7z, maybe not safe to assume is always archive name minus extension.

**Comment by fuzzard** [comment]
Indeed. just need the time :)

**Pull request merged**